### ISSUE TITLE
Intermittent segfault at process exit: library unloads while libdatachannel's detached cleanup thread is still running

### ISSUE THREAD

**Issue by anthonyf** [godotengine/webrtc-native#195]
**Godot version:** 4.7.stable (also reproducible setup on 4.7.1)
**webrtc-native version:** 1.2.1-stable (GDExtension, libdatachannel 0.24.5)
**Platform:** Linux x86_64 (GitHub Actions `ubuntu-latest` runners; not yet observed on macOS)

## Symptom

A headless Godot process that creates and destroys a few `WebRTCPeerConnection` instances (never connected — just `new()` + `close()` + free) intermittently dies with `Segmentation fault (core dumped)` **during engine shutdown**, after the script has finished and called `SceneTree.quit(0)`. All output is printed; the crash happens ~0.5s later during cleanup. Roughly 1 in a few hundred runs on 2-core CI runners.

## Cause

`WebRTCLibPeerConnection`'s constructor creates a real `rtc::PeerConnection`, which acquires libdatachannel's global init token. When the **last** peer connection is destroyed, libdatachannel's `Init::TokenPayload::~TokenPayload()` spawns a **detached** "RTC cleanup" thread that runs `Init::doCleanup()` (thread-pool join, `usrsctp` teardown, etc.):

https://github.com/paullouisageneau/libdatachannel/blob/v0.24.5/src/impl/init.cpp — `~TokenPayload()` uses `std::thread(...).detach()`.

At extension deinit, `WebRTCLibPeerConnection::deinitialize_signaling()` calls:

```cpp
void WebRTCLibPeerConnection::deinitialize_signaling() {
	rtc::Cleanup();
}
```

But `rtc::Cleanup()` is asynchronous — its signature is `std::shared_future<void> Cleanup();` and the returned future (completed by the detached cleanup thread) is discarded. Godot then proceeds to `dlclose` the extension library. If the detached thread is still executing code inside `libwebrtc_native.so` when the library is unmapped, the process segfaults. It's a race, so it only bites occasionally, and more often on slow/oversubscribed machines (`usrsctp_finish` in particular can take a while).

This looks like the missing half of #132, which added the `rtc::Cleanup()` call but doesn't wait for it to finish.

## Suggested fix

Wait for the cleanup future before returning from `deinitialize_signaling()`, so the library is never unloaded while the cleanup thread is inside it:

```cpp
void WebRTCLibPeerConnection::deinitialize_signaling() {
	rtc::Cleanup().wait();
}
```

(Or `wait_for(...)` with a generous timeout, if blocking indefinitely at shutdown is a concern.)

## Reproduction

Headless script that creates a few `WebRTCPeerConnection` instances, closes/frees them, and quits — looped a few hundred times on a Linux runner:

- Failing CI run (test passes, then segfaults in teardown): https://github.com/anthonyf/umbral-vtt/actions/runs/30725843573
- Loop reproduction on `ubuntu-latest` (2 shards × 400 iterations): https://github.com/anthonyf/umbral-vtt/actions/runs/30726849538 — **121/400 and 91/400 runs segfault** after the script completes. An otherwise-identical variant whose peer connections are script-only stubs (`extends WebRTCPeerConnectionExtension`, so no libdatachannel object is ever constructed): **0/800**.

Core-dump backtrace from a crashed run (gdb, Godot 4.7-stable official Linux x86_64). The segfaulting thread and the thread `pthread_join`ing it are both executing at addresses gdb cannot attribute to any mapped library — the region where the extension was loaded before `dlclose`:

```
Thread 4 (LWP 3639):
#3  __pthread_clockjoin_ex (threadid=..., ...) at ./nptl/pthread_join_common.c:102
#4  0x00007fa060b731f9 in ?? ()          <- unmapped extension region
#5  0x00007fa060e27f50 in ?? ()
#6  0x00007fa060b42217 in ?? ()

Thread 1 (LWP 3638) — crashed:
#4  <signal handler called>
#5  0x00007fa060b44ba5 in ?? ()          <- SIGSEGV here, unmapped extension region
```

**Comment by Faless** [comment]
Thanks for the report, I've addressed this in #196 which will be part of the next release.
